modal/running_app.py

```diff
@@ -4,16 +4,42 @@
 
 from google.protobuf.message import Message
 
+from modal._utils.grpc_utils import get_proto_oneof
+from modal_proto import api_pb2
+
 from .client import _Client
 
 
 @dataclass
 class RunningApp:
     app_id: str
+    client: _Client
     environment_name: Optional[str] = None
     app_page_url: Optional[str] = None
     app_logs_url: Optional[str] = None
     tag_to_object_id: dict[str, str] = field(default_factory=dict)
     object_handle_metadata: dict[str, Optional[Message]] = field(default_factory=dict)
     interactive: bool = False
-    client: Optional[_Client] = None
+
+
+def running_app_from_layout(
```

Comment:
this helper function is probably a slightly unnecessary layer of indirection. We should be able to simplify things later by making `RunningApp` look more like `api_pb2.AppLayout`